provider/parameter.go

```diff
@@ -21,6 +21,25 @@ import (
 	"golang.org/x/xerrors"
 )
 
+type ValidationMode string
+
+const (
+	ValidationModeEnvVar = "CODER_VALIDATION_MODE"
+	// ValidationModeDefault is used for creating a workspace. It validates the final
+	// value used for a parameter. Some allowances for invalid options are tolerated,
+	// as unused options do not affect the final parameter value. The default value
+	// is also ignored, assuming a value is provided.
+	ValidationModeDefault ValidationMode = ""
+	// ValidationModeTemplateImport tolerates empty values, as the value might not be
+	// available at import. It does not tolerate an invalid default or invalid option
```

Comment:
IIUC, It doesn't tolerate empty values if there is a validation defined, so this is misleading.

Reply:
Correct. I will update